### DISCUSSION TITLE
DALL-E-3 on Azure stop working?

### DISCUSSION THREAD

**Discussion by cnkang** [danny-avila/LibreChat#2722] [Troubleshooting] [answered]
### What happened?

When attempting to generate an image using DALL-E-3, I encountered a 404 error. I haven't made any changes to the settings, but I did upgrade to the latest code a few days ago. I remember it was working fine before the upgrade. I tested the endpoint and API key with the 'curl' command, and they worked fine. Appreciate your assistance.

Output from the container:
LibreChat | 2024-05-15 12:01:56 error: [DALL-E-3] Problem generating the image: 404 Resource not found

From the browser:
RESPONSE FROM DALL-E-3
Something went wrong when trying to generate the image. The DALL-E API may be unavailable:
Error Message: 404 Resource not found

My settings:
DALLE3_API_KEY=my-key
DALLE3_AZURE_API_VERSION=2024-02-1
DALLE3_BASEURL=https://myname-test-swedencentral.openai.azure.com/openai/deployments/dall-e-3/

Curl command output:
% curl -X POST \
     -H "api-key: my-key" \
     -H "Content-Type: application/json" \
     -d '{"prompt":"A multi-colored umbrella on the beach, disposable camera","size":"1024x1024","n":1,"quality":"hd","style":"vivid"}' \
     'https://myname-test-swedencentra.openai.azure.com/openai/deployments/dall-e-3/images/generations?api-version=2024-02-01'

{"created":1715773752,"data":[{"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"prompt_filter_results":{"hate":{"filtered":false,"severity":"safe"},"profanity":{"detected":false,"filtered":false},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"revised_prompt":"Visualize a scene at a sandy beach on a sunny day. A magnificently multi-colored umbrella stands firmly in the sand, providing much-needed shade against the bright sunlight. A few steps away is a vintage disposable camera, its worn out exterior suggesting countless memories captured within its simple frame. Gentle waves lap against the shores in the background, and footprints mark the passage of people. The whole scene exudes warmth, relaxation, and promises of memorable moments under the colorfully striped beach umbrella.","url":"...."

### Steps to Reproduce

1. ask DALL-E-3 to generate a image

### Relevant log output

```shell
LibreChat         | 2024-05-15 12:01:56 error: [DALL-E-3] Problem generating the image: 404 Resource not found
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cnkang** [comment, marked as answer]
sorry, my bad. someone changed the api-version with a wrong value, that trigger the 404 error.
after I correct the value, it works.

**Reply by danny-avila** [reply]
No worries, thank you for confirming.
